assets/js/googlesitekit/site/index.js

```diff
@@ -0,0 +1,210 @@
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import { get, set } from 'googlesitekit-api';
+import {
+	FETCH_CONNECTION_INFO,
+	FETCH_RESET,
+	INITIALIZE,
+	RESET,
+	RESET_FAILURE,
+	RESET_SUCCESS,
+	RECEIVE_CONNECTION_INFO,
+	RECEIVE_CONNECTION_INFO_FAILED,
+} from './index.private';
+
+export const INITIAL_STATE = {
+	connectionInfo: null,
+	isFetchingConnectionInfo: false,
+	isResetting: false,
+};
+
+export const STORE_NAME = 'core/site';
+
+export const actions = {
+	fetchConnectionInfo() {
+		return {
+			payload: {},
+			type: FETCH_CONNECTION_INFO,
+		};
+	},
+
+	fetchReset() {
+		return {
+			payload: {},
+			type: FETCH_RESET,
+		};
+	},
+
+	initialize() {
+		return {
+			payload: {},
+			type: INITIALIZE,
+		};
+	},
+
+	receiveConnectionInfo( connectionInfo ) {
+		invariant( connectionInfo, 'connectionInfo is required.' );
+
+		return {
+			payload: { connectionInfo },
+			type: RECEIVE_CONNECTION_INFO,
+		};
+	},
+
+	receiveConnectionInfoFailed( error ) {
+		invariant( error, 'error is required.' );
+
+		return {
+			payload: { error },
+			type: RECEIVE_CONNECTION_INFO_FAILED,
+		};
+	},
+
+	*reset() {
+		try {
+			yield actions.fetchReset();
+			yield actions.resetSuccess();
+			yield actions.initialize();
+
+			return {
+				payload: {},
+				type: RESET,
+			};
+		} catch ( err ) {
+			return actions.resetFailure( err );
+		}
+	},
+
+	resetFailure( error ) {
+		invariant( error, 'error is required.' );
+		return {
+			payload: { error },
+			type: RESET_FAILURE,
+		};
+	},
+
+	resetSuccess() {
```

Comment:
Although `resetFailure` and `resetSuccess` are based on an action and not a selector, we should still stick with the convention and call them `receiveResetFailed` and `receiveReset`.

In other words, any "non-public" action should be called `fetch...` or `receive...`.

Reply:
Ah, I like that. 👍